--- src/bitmap.rs
+++ src/bitmap.rs
@@ -296,1 +296,1 @@
-        let mut iter = bools.array_chunks::<BITS>();
+        let mut iter = bools.iter().copied().array_chunks::<BITS>();
@@ -299,1 +299,1 @@
-            let bitmask = std::simd::Mask::<i8, BITS>::from_array(*chunk).to_bitmask() as usize;
+            let bitmask = std::simd::Mask::<i8, BITS>::from_array(chunk).to_bitmask() as usize;
@@ -302,1 +302,1 @@
-        if !iter.remainder().is_empty() {
+        if let Some(remainder) = iter.into_remainder() {
@@ -304,2 +304,2 @@
-            for (i, b) in iter.remainder().iter().enumerate() {
-                bitmask |= (*b as usize) << i;
+            for (i, b) in remainder.enumerate() {
+                bitmask |= (b as usize) << i;

--- src/lib.rs
+++ src/lib.rs
@@ -29,1 +29,1 @@
-#![feature(array_chunks)]
+#![feature(iter_array_chunks)]

--- src/lru.rs
+++ src/lru.rs
@@ -170,1 +170,1 @@
-                    let ptr = NonNull::new_unchecked(Box::into_raw(entry));
+                    let ptr = NonNull::new_unchecked(Box::into_raw_with_allocator(entry).0);

--- src/row/mod.rs
+++ src/row/mod.rs
@@ -314,1 +314,1 @@
-            self.as_ref().len()
+            self.len()
@@ -319,1 +319,1 @@
-            self.as_ref().iter().map(ToDatumRef::to_datum_ref)
+            AsRef::<[D]>::as_ref(self).iter().map(ToDatumRef::to_datum_ref)