--- src/binder/expr/function/builtin_scalar.rs
+++ src/binder/expr/function/builtin_scalar.rs
@@ -228,1 +228,3 @@
-                // string
+                // marrow
+                ("add_one", raw_call(ExprType::AddOne)),
+                // string

--- src/catalog/mod.rs
+++ src/catalog/mod.rs
@@ -110,1 +110,1 @@
-    #[provide(PostgresErrorCode => PostgresErrorCode::UndefinedObject)]
+    //#[extra_provide(PostgresErrorCode => PostgresErrorCode::UndefinedObject)]

--- src/catalog/mod.rs
+++ src/catalog/mod.rs
@@ -114,1 +114,1 @@
-    #[provide(PostgresErrorCode => PostgresErrorCode::DuplicateObject)]
+    //#[extra_provide(PostgresErrorCode => PostgresErrorCode::DuplicateObject)]

--- src/expr/pure.rs
+++ src/expr/pure.rs
@@ -284,1 +284,2 @@
-            | Type::CompositeCast =>
+            | Type::CompositeCast
+            | Type::AddOne =>

--- src/optimizer/plan_expr_visitor/strong.rs
+++ src/optimizer/plan_expr_visitor/strong.rs
@@ -354,1 +354,2 @@
-            | ExprType::Random => false,
+            | ExprType::Random
+            | ExprType::AddOne => false,